packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -138,10 +138,31 @@ export interface UserPoolTriggers {
    */
   readonly verifyAuthChallengeResponse?: lambda.IFunction;
 
+  /**
+   * Amazon Cognito invokes this trigger to send email notifications to users.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-custom-email-sender.html
+   * @default - no trigger configured
+   */
+  readonly customEmailSender?: lambda.IFunction
+
+  /**
+   * Amazon Cognito invokes this trigger to send SMS notifications to users.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-custom-sms-sender.html
+   * @default - no trigger configured
+   */
+  readonly customSmsSender?: lambda.IFunction
+
+  /**
+   * This key will be used to encrypt temporary passwords and authorization codes that Amazon Cognito generates.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-custom-sender-triggers.html
+   * @default - no key ID configured
+   */
+  readonly kmsKeyId?: string
+
   /**
    * Index signature
    */
-  [trigger: string]: lambda.IFunction | undefined;
+  [trigger: string]: lambda.IFunction | string | undefined;
```

Comment:
union types such as these are not supported by jsii.

In Java, they will get incorrectly translated into `java.lang.Object`.

Move the kms key to the top-level `UserPoolProps`.
Call it `customSenderKmsKey` with type `IKey` from the `@aws-cdk/aws-kms` module?